### DISCUSSION TITLE
From Subtitle轉入SSML的duration 參數 及 SSML中prosody 控制的問題

### DISCUSSION THREAD

**Discussion by hymch** [lenML/Speech-AI-Forge#151] [问答 | Q&A]
使用 From Subtitle 加載 字幕文件轉入SSML模塊後
如果改動 
<voice duration=4000ms spk=".......> 中的duration 參數
將會改變語速，但這改變並不符合直覺
縮短duration 會令語速減慢，加大duration反而令語速加快
而且如此加減速對pitch的影響比使用调节器要小

此外 SSML使用 prosody 控制無效
本地和HF space都復現，文本如下:
```
<speak version="0.1">
    <voice spk="Bob" seed="42" style="narration-relaxed">
        使用 prosody 控制生成文本的语速语调和音量，示例如下 [lbreak]
        <prosody rate="0.5">
            设置 rate 大于1表示加速，小于1为减速 [lbreak]
        </prosody>
    </voice>
</speak>
```
後句語速不變
但用TTS模塊的调节器測試 速度=0.5 確可以調節

謝謝

**Comment by hymch** [comment]
另外 有個 From Subtitle 的想法，
希望能在導入字幕文件後，加入表格式的參數註釋，方才輸出到SSML模塊
現在From Subtitle頁面只有音色選擇，其餘參數需在輸出後修改，較為無序

**Reply by zhzLuke96** [reply]
from subtitle 这块属于尝试性的加了一个演示功能，探索下ssml的用法，所以也还是比较粗糙的

我大概理解你需要的是类似 podcast tab 里面的那种表格，可以调整文本、说话人？

**Reply by hymch** [reply]
> 我大概理解你需要的是类似 podcast tab 里面的那种表格，可以调整文本、说话人？

是的, 希望在from subtitle 裡就能直接表格輸入style, rate 之類會比較系統

**Reply by hymch** [reply]
呀 另外想問下，未來有考慮讓SSML兼容其他模型比如cosyvoice嗎?

**Reply by zhzLuke96** [reply]
> 呀 另外想問下，未來有考慮讓SSML兼容其他模型比如cosyvoice嗎?

现在就可以
没有什么兼容不兼容的，这个推理架构下面模型调用实现忘了，所有功能都通用 （人声增强、调速、openvoice以及ssml）
你要是用 api 其他模型一样也可以用 ssml 调用，webui只是暂时没空搞还没写切换模型的功能

**Comment by zhzLuke96** [comment]
语速这个问题是bug，一会我看看
- https://github.com/lenML/ChatTTS-Forge/issues/152

**Reply by hymch** [reply]
呀這 那這可以修成一個功能嗎, 感覺這樣調的效果還不錯的 XD

**Reply by zhzLuke96** [reply]
> 呀這 那這可以修成一個功能嗎, 感覺這樣調的效果還不錯的 XD

这个duration属性是会保留的，只是现在是反了的嘛，不是吗

**Comment by zhzLuke96** [comment]
更新一下：
- duration 控制问题已经修复
- webui 中 SSML 下增加了一个 `Script` 脚本工具，从其他tab中可以发送结果到脚本或者发送到 SSML，脚本页面以表格形式修改调整片段
